File: 2021/May/Day 27/kthfactor.cpp
// Source : https://leetcode.com/problems/the-kth-factor-of-n/submissions/
class Solution {
public:
    int kthFactor(int n, int k) {

        vector<int> f;

        for(int i=1;i<=n;i++)
        {
            if(n%i==0)
            {
                f.insert(f.end(),i);
            }
        }

        if(size(f)>=k)
        {
            return f[k-1];
        }
        return -1;

    }
};

File: 2021/May/Day 27/kthfactor.py
# Source : https://leetcode.com/problems/the-kth-factor-of-n/submissions/
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        factor = [i for i in range(1,n+1) if n%i==0]
        if(len(factor)>=k):
            return factor[k-1]
        return -1
